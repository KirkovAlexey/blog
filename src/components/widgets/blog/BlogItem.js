import React, { DOM } from 'react';
import PropTypes from 'prop-types';

import TextBox from './elements/TextBox';
import Image from './elements/Image';
import Like from './elements/Like';
import MetaData from './elements/Meta';
import ButtonEditPost from './elements/ButtonEditPost';
import Link from 'components/elements/Link';

import moment from 'moment';

import { Divider, Grid } from 'semantic-ui-react';

const BlogItem = ({ element, handleLikeClick }) => (
  React.createElement(
    Grid,
    {},
    React.createElement(
      Grid.Row,
      {},
      React.createElement(
        Grid.Column,
        { width: 3 },
        React.createElement(Image, element.image)
      ),
      React.createElement(
        Grid.Column,
        { width: 8 },
        React.createElement(
          'p',
          {},
          React.createElement(
            Link,
            { to: element.link },
            element.description.text
          )
        ),
        React.createElement(
          ButtonEditPost,
          { id: element.id }
        ),
        handleLikeClick && React.createElement(
          Like,
          {
            meta: element.meta,
            id: element.id,
            handleLikeClick
          }
        ),
        React.createElement('hr', {}),
        React.createElement(MetaData, element.meta)
      )
    ),
    React.createElement(Divider, { horizontal: true })
  )
);

BlogItem.propTypes = {
  element: PropTypes.object,
  handleLikeClick: PropTypes.func
};

BlogItem.defaultProps = {
  element: {
    description: {
      text: PropTypes.string
    }
  }
};

export default BlogItem;
