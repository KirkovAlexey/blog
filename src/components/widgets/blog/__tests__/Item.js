import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Link from 'components/elements/Link';

import BlogItem from '../BlogItem';

describe('Item', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BlogItem />, div);
  });

  describe('render', () => {
    it('should render the title', () => {
      const elementProps = {
        link: '/posts/5',
        description: { text: 'Hello, world!' }
      };
      const item = shallow(<BlogItem element={elementProps} />);
      const header = <Link to="/posts/5">Hello, world!</Link>;

      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const elementProps = {
        link: '/posts/5',
        description: { text: 'Hello, world!' },
        image: {
          alt: 'Hello, World',
          src: '/dist/images/default.jpg'
        }
      };

      const item = shallow(<BlogItem element={elementProps} />);

      expect(item).toMatchSnapshot();
    });
  });
});
