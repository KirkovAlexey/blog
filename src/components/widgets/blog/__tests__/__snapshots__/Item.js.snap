// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Grid>
    <GridRow>
        <GridColumn
            width={3}
        >
            <Image
                alt="Hello, World"
                height={120}
                src="/dist/images/default.jpg"
                style={
                    Object {
                        "border": "1px solid red",
                        "margin": "5px",
                        "padding": "5px",
                      }
                }
                width={160}
            />
        </GridColumn>
        <GridColumn
            width={8}
        >
            <p>
                <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/posts/5"
                >
                    Hello, world!
                </Link>
            </p>
            <ButtonEditPost
                id={undefined}
            />
            <hr />
            <MetaData
                author="Lost autor"
                createdAt="Today at 5:51 PM"
                updatedAt="Today at 5:51 PM"
            />
        </GridColumn>
    </GridRow>
    <Divider
        horizontal={true}
    />
</Grid>,
  "nodes": Array [
    <Grid>
      <GridRow>
            <GridColumn
                  width={3}
            >
                  <Image
                        alt="Hello, World"
                        height={120}
                        src="/dist/images/default.jpg"
                        style={
                              Object {
                                    "border": "1px solid red",
                                    "margin": "5px",
                                    "padding": "5px",
                                  }
                        }
                        width={160}
                  />
            </GridColumn>
            <GridColumn
                  width={8}
            >
                  <p>
                        <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/posts/5"
                        >
                              Hello, world!
                        </Link>
                  </p>
                  <ButtonEditPost
                        id={undefined}
                  />
                  <hr />
                  <MetaData
                        author="Lost autor"
                        createdAt="Today at 5:51 PM"
                        updatedAt="Today at 5:51 PM"
                  />
            </GridColumn>
      </GridRow>
      <Divider
            horizontal={true}
      />
</Grid>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        element={
                Object {
                        "description": Object {
                          "text": "Hello, world!",
                        },
                        "image": Object {
                          "alt": "Hello, World",
                          "src": "/dist/images/default.jpg",
                        },
                        "link": "/posts/5",
                      }
        }
/>,
      "_debugID": 34,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "element": Object {
            "description": Object {
              "text": "Hello, world!",
            },
            "image": Object {
              "alt": "Hello, World",
              "src": "/dist/images/default.jpg",
            },
            "link": "/posts/5",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Grid>
          <GridRow>
                    <GridColumn
                              width={3}
                    >
                              <Image
                                        alt="Hello, World"
                                        height={120}
                                        src="/dist/images/default.jpg"
                                        style={
                                                  Object {
                                                            "border": "1px solid red",
                                                            "margin": "5px",
                                                            "padding": "5px",
                                                          }
                                        }
                                        width={160}
                              />
                    </GridColumn>
                    <GridColumn
                              width={8}
                    >
                              <p>
                                        <Link
                                                  onlyActiveOnIndex={false}
                                                  style={Object {}}
                                                  to="/posts/5"
                                        >
                                                  Hello, world!
                                        </Link>
                              </p>
                              <ButtonEditPost
                                        id={undefined}
                              />
                              <hr />
                              <MetaData
                                        author="Lost autor"
                                        createdAt="Today at 5:51 PM"
                                        updatedAt="Today at 5:51 PM"
                              />
                    </GridColumn>
          </GridRow>
          <Divider
                    horizontal={true}
          />
</Grid>,
        "_debugID": 35,
        "_renderedOutput": <Grid>
          <GridRow>
                    <GridColumn
                              width={3}
                    >
                              <Image
                                        alt="Hello, World"
                                        height={120}
                                        src="/dist/images/default.jpg"
                                        style={
                                                  Object {
                                                            "border": "1px solid red",
                                                            "margin": "5px",
                                                            "padding": "5px",
                                                          }
                                        }
                                        width={160}
                              />
                    </GridColumn>
                    <GridColumn
                              width={8}
                    >
                              <p>
                                        <Link
                                                  onlyActiveOnIndex={false}
                                                  style={Object {}}
                                                  to="/posts/5"
                                        >
                                                  Hello, world!
                                        </Link>
                              </p>
                              <ButtonEditPost
                                        id={undefined}
                              />
                              <hr />
                              <MetaData
                                        author="Lost autor"
                                        createdAt="Today at 5:51 PM"
                                        updatedAt="Today at 5:51 PM"
                              />
                    </GridColumn>
          </GridRow>
          <Divider
                    horizontal={true}
          />
</Grid>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    element={
        Object {
            "description": Object {
              "text": "Hello, world!",
            },
            "image": Object {
              "alt": "Hello, World",
              "src": "/dist/images/default.jpg",
            },
            "link": "/posts/5",
          }
    }
/>,
}
`;
